File: parking-domain/src/main/java/com/chatchatabc/parking/impl/domain/event/UserEventProcessorImpl.java
package com.chatchatabc.parking.impl.domain.event;

import com.chatchatabc.parking.domain.event.user.UserEventProcessor;
import com.chatchatabc.parking.domain.event.user.UserLoginEvent;
import com.chatchatabc.parking.impl.infra.schedule.user.UserSchedule;
import org.quartz.SchedulerException;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.event.EventListener;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Component;

@Component
public class UserEventProcessorImpl implements UserEventProcessor {
    @Autowired
    private UserSchedule userSchedule;

    /**
     * Handle user login event
     *
     * @param event the user login event
     */
    @Async
    @EventListener
    @Override
    public void handleUserLoginEvent(UserLoginEvent event) throws SchedulerException {
        String phone = event.getPhone();
        String otp = event.getOtp();
        userSchedule.onLoginSendSMSOTP(phone, otp);
    }
}

File: parking-api/src/main/kotlin/com/chatchatabc/parking/api/impl/application/event/UserEventProcessorImpl.kt
package com.chatchatabc.parking.api.impl.application.event

import com.chatchatabc.parking.api.application.event.user.UserEventProcessor
import com.chatchatabc.parking.api.application.event.user.UserLoginEvent
import com.chatchatabc.parking.api.application.schedule.user.UserSchedule
import org.springframework.context.event.EventListener
import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Component

@Component
class UserEventProcessorImpl(
    private val userSchedule: UserSchedule
) : UserEventProcessor {
    /**
     * Handle user login event
     */
    @Async
    @EventListener
    override fun handleUserLoginEvent(event: UserLoginEvent) {
        val phone = event.phone
        val otp = event.otp
        userSchedule.onLoginSendSMSOTP(phone, otp)
    }
}